assets/js/googlesitekit/datastore/modules/modules.js

```diff
@@ -0,0 +1,434 @@
+
+		if ( ! existingModules ) {
+			yield actions.fetchModules();
+		}
+	},
+};
+
+export const selectors = {
+	/**
+	 * Get the list of modules registered for use with Site Kit.
+	 *
+	 * A module is a section of Site Kit that relates to a particular service,
+	 * like Google Analytics or Google PageSpeed modules. They can provide
+	 * admin-only features (like PageSpeed Insights), frontend-only features,
+	 * or both (eg. Analytics, which can install Analytics <script> tags in the
+	 * frontend, and show dashboards in the WordPress Admin).
+	 *
+	 * Returns a hash of objects with the shape, keyed by slug, when successful:
+	 * ```
+	 * slug: {
+	 *   "slug": "tagmanager",
+	 *   "name": "Tag Manager",
+	 *   "description": "Tag Manager creates an easy to manage way to create tags on your site without updating code.",
+	 *   "homepage": "https://tagmanager.google.com/",
+	 *   "internal": false,
+	 *   "active": false,
+	 *   "connected": false,
+	 *   "dependencies": [
+	 *     "analytics"
+	 *   ],
+	 *   "dependants": []
+	 * }
+	 * ```
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object|undefined} Modules available on the site.
+	 */
+	getModules( state ) {
+		const { modules } = state;
+
+		return modules;
+	},
+
+	/**
+	 * Get a specific module by slug.
+	 *
+	 * Returns a specific module by its slug.
+	 * Returns `undefined` if state is still loading or if said module doesn't exist.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Module slug.
+	 * @return {Object|undefined} A specific module object.
+	 */
+	getModule: createRegistrySelector( ( select ) => ( state, slug ) => {
+		const modules = select( STORE_NAME ).getModules();
+
+		// Return `undefined` if modules haven't been loaded yet.
+		if ( modules === undefined ) {
+			return undefined;
+		}
+
+		// A module with this slug couldn't be found; return `null` to signify the
+		// "not found" state.
+		if ( modules[ slug ] === undefined ) {
+			return null;
+		}
+
+		// This module exists, so let's return it.
+		return modules[ slug ];
+	} ),
+
+	/**
+	 * Check a module's activation status.
+	 *
+	 * Returns `true` if the module exists and is active.
+	 * Returns `false` if the module exists but is not active.
+	 * Returns `undefined` if state is still loading or if no module with that slug exists.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} slug  Module slug.
+	 * @return {Object|undefined} A specific module object.
+	 */
+	isModuleActive: createRegistrySelector( ( select ) => ( state, slug ) => {
+		const module = select( STORE_NAME ).getModule( slug );
+
+		// Return `undefined` if modules haven't been loaded yet.
+		if ( module === undefined ) {
+			return undefined;
+		}
+
+		// A module with this slug couldn't be found; return `null` to signify the
+		// "not found" state.
+		if ( module === null ) {
+			return null;
```

Comment:
I guess that makes sense here - I'm _a bit_ questioning whether we should just return `false`, to simplify the signature here. Logically, if a module doesn't exist, it's also not active, so returning `false` could still be appropriate.

Reply:
I suppose so, but I like clarifying the reason for the value here. Code that doesn't care if it exists or not can always check for a truthy value, but this exposes a bit more context for those components that might care about _why_ it's returning false-y without needing other selectors.